spotify_dl/spotify.py

```diff
@@ -13,110 +14,200 @@ def fetch_tracks(sp, item_type, url):
+                progress.update(
+                    task_id=album_songs_task,
+                    description=f"Fetched {offset} of {album_total} songs from the album {track_album}",
+                    advance=offset,
+                    total=album_total,
+                )
                 if album_total == offset:
                     break
 
-    elif item_type == 'track':
+    elif item_type == "track":
         items = sp.track(track_id=url)
-        track_name = items.get('name')
-        album_info = items.get('album')
-        track_artist = ", ".join([artist['name'] for artist in items['artists']])
+        track_name = items.get("name")
+        album_info = items.get("album")
+        track_artist = ", ".join([artist["name"] for artist in items["artists"]])
         if album_info:
-            track_album = album_info.get('name')
-            track_year = album_info.get('release_date')[:4] if album_info.get('release_date') else ''
-            album_total = album_info.get('total_tracks')
-        track_num = items['track_number']
-        spotify_id = items['id']
-        if len(items['album']['images']) > 0:
-            cover = items['album']['images'][0]['url']
+            track_album = album_info.get("name")
+            track_year = (
+                album_info.get("release_date")[:4]
+                if album_info.get("release_date")
+                else ""
+            )
+            album_total = album_info.get("total_tracks")
+        track_num = items["track_number"]
+        spotify_id = items["id"]
+        if len(items["album"]["images"]) > 0:
+            cover = items["album"]["images"][0]["url"]
         else:
             cover = None
-        if len(sp.artist(artist_id=items['artists'][0]['uri'])['genres']) > 0:
-            genre = sp.artist(artist_id=items['artists'][0]['uri'])['genres'][0]
+        if len(sp.artist(artist_id=items["artists"][0]["uri"])["genres"]) > 0:
+            genre = sp.artist(artist_id=items["artists"][0]["uri"])["genres"][0]
         else:
             genre = ""
-        songs_list.append({"name": track_name, "artist": track_artist, "album": track_album, "year": track_year,
-                           "num_tracks": album_total, "num": track_num, "playlist_num": offset + 1,
-                           "cover": cover, "genre": genre, "spotify_id": spotify_id})
+        songs_list.append(
+            {
+                "name": track_name,
+                "artist": track_artist,
+                "album": track_album,
+                "year": track_year,
+                "num_tracks": album_total,
+                "num": track_num,
+                "playlist_num": offset + 1,
+                "cover": cover,
+                "genre": genre,
+                "spotify_id": spotify_id,
+            }
+        )
```

Comment:
most of the formatting in the code , i used black formatter `pip install black` if interested

Reply:
please add them to dev requirements, as well as any custom black configs that you've used

Reply:
also set the width to 120

Reply:
is it the line length i should set to 120. just read through the bllack formatter docs and the best i git is making a pyproject.toml file and addind the 120 length file setting .